pybamm/expression_tree/unary_operators.py

```diff
@@ -172,19 +172,19 @@ class NumpyBroadcast(Broadcast):
         child node
     domain : iterable of string
         the domain to broadcast the child to
-    mesh : mesh class
-        the mesh used for discretisation
+    npts : dict
+        contains the number of points on each domain
 
     **Extends:** :class:`SpatialOperator`
     """
 
-    def __init__(self, child, domain, mesh):
+    def __init__(self, child, domain, npts):
         super().__init__(child, domain, name="numpy broadcast")
         # determine broadcasting vector size (size 1 if the domain is empty)
         if domain == []:
             self.broadcasting_vector_size = 1
         else:
-            self.broadcasting_vector_size = sum([mesh[dom].npts for dom in domain])
+            self.broadcasting_vector_size = sum([npts[dom] for dom in domain])
```

Comment:
This seems like an unnecessary change that obscures what the class is doing, and requires you to artificially create the dictionary `npts` before calling `NumpyBroadcast`. I think that the `npts` attribute of a submesh should always be defined in such a way that `mesh[dom].npts` is what you want to use for broadcasting

Reply:
The choice is to:
1) change `SubMesh` so that we add  .npts to each SubMesh within `SpatialMethod` and then pass the new mesh
or 
2) create a dict of npts in each domain and change broadcast to take in npts dict 

I didn't want to change the `SubMesh` within a spatial method so that the same property names refer to slightly different things for different spatial methods. It seems clearer to me to have a submesh.cells and a submesh.edges property rather than renaming one of them submesh.nodes within the a spatial method. Likewise, having submesh.n_cells and a submesh.n_edges instead of submesh.npts.

Reply:
ok so rename `nodes` to `cells` and `npts` to `n_cells` and then use `mesh[dom].n_cells`?